ntoskrnl/cc/view.c

```diff
@@ -497,13 +497,21 @@ CcRosTrimCache(
         /* Check if it's mapped and not dirty */
         if (InterlockedCompareExchange((PLONG)&current->MappedCount, 0, 0) > 0 && !current->Dirty)
         {
+            /* We have to break these locks to call MmPageOutPhysicalAddress */
+            KeReleaseSpinLockFromDpcLevel(&current->SharedCacheMap->CacheMapLock);
+            KeReleaseQueuedSpinLock(LockQueueMasterLock, oldIrql);
```

Comment:
Breaking this lock will make any pointer to VACBs in `VacbLruListHead` potentially invalid. While there is a reference on `current`, preventing it from beig deleted, there is no reference on `current_entry`. This means after breaking the lock, you have to either start over at `VacbLruListHead.Flink`, which can potentially lead to an infinite loop. or take a reference on the next VACB.

Reply:
@HBelusca, Here is where Timo says that this statement should be moved. Thanks.